pintc/src/predicate/transform/lower.rs

```diff
@@ -1037,15 +1037,15 @@ pub(crate) fn lower_ifs(contract: &mut Contract) {
 
         // Ideally we'd refactor this to not require cloning the IfDecl.
         for if_decl in &contract.preds[pred_key].if_decls.clone() {
-            all_exprs.extend(&convert_if(contract, pred_key, if_decl));
+            all_exprs.extend(convert_if(contract, pred_key, if_decl));
```

Comment:
From my limited understanding this is still functionally the same because of how `.extends()` handles a return type of `Vec<ExprKey>` and `Vec<(ExprKey, Span)>`

Reply:
Did you need to remove the `&` reference for it to compile?  `extend()` takes any type which implements `IntoIterator` and the result of `convert_if()` is a temporary rvalue so I'm surprised it makes any difference.

Taking the `&` returns a slice which would then be owned by the iterator and just passing the value would mean the `Vec` itself is owned by the iterator but either should work.  The only difference is if you wanted to use the result of `convert_if()` later on, since it's moved into `extend()`.

Reply:
I did need to remove the `&`

When I keep the `&`, rust-analyzer says I need to dereference `expr` when pushing `ConstraintDecl`

```rust
for if_decl in &contract.preds[pred_key].if_decls.clone() {
    all_exprs.extend(&convert_if(contract, pred_key, if_decl));
}

let mut_pred = contract.preds.get_mut(pred_key).unwrap();

for (expr, original_expr_span) in all_exprs {
    mut_pred.constraints.push(ConstraintDecl {
        expr: *expr,
        span: original_expr_span,
    });
}
```

Which makes sense to me. But once I do that, I get 
>temporary value dropped while borrowed
>consider using a `let` binding to create a longer lived value

for `&convert_if(contract, pred_key, if_decl)` which I believe makes the use of `&` pointless here because I would need to hold on to the values anyway?


FWIW when I use `&`, `all_exprs` is of type `Vec<&(ExprKey, Span)>`, and when I don't use `&`, `all_exprs` is of type `Vec<(ExprKey, Span)>`. Which reflects what `all_exprs` used to be (`Vec<ExprKey>`) before `convert_if` returned a `tuple`.

Reply:
Happy to change it of course. Just let me know what you think and I can make a quick PR to modify :D